code/modules/mob/living/carbon/xenomorph/damage_procs.dm

```diff
@@ -173,3 +173,25 @@
 		else
 			if(decal.random_icon_states && length(decal.random_icon_states) > 0) //If there's already one, just randomize it so it changes.
 				decal.icon_state = pick(decal.random_icon_states)
+
+		var/splash_chance = 40 //Base chance of getting splashed. Decreases with # of victims.
+		var/distance = 0 //Distance, decreases splash chance.
+		var/i = 0 //Tally up our victims.
+		
+		if(!(xeno_caste.caste_flags & CASTE_ACID_BLOOD))
+			return
+		else
+			for(var/mob/living/carbon/human/victim in range(radius,src)) //Loop through all nearby victims, including the tile.
+				distance = get_dist(src,victim)
+
+				splash_chance = 80 - (i * 5)
+				if(victim.loc == loc) splash_chance += 30 //Same tile? BURN
+				splash_chance += distance * -15
+
+				if(splash_chance > 0 && prob(splash_chance)) //Success!
```

Comment:
`prob` of a negative or zero value will return false, so no need to check for if it's higher than zero.